packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -98,16 +98,41 @@ Future<GradleProject> _readGradleProject() async {
     return project;
   } catch (e) {
     if (flutterPluginVersion == FlutterPluginVersion.managed) {
-      printError('Error running Gradle:\n$e\n');
-      throwToolExit(
-        'Please review your Gradle project setup in the android/ folder.',
-      );
+      // Handle known exceptions. This will exit if handled.
+      handleKnownGradleExceptions(e);
+      
+      // Print a general Gradle error and exit.
+      printError('* Error running Gradle:\n$e\n');
+      throwToolExit('Please review your Gradle project setup in the android/ folder.');
     }
   }
   // Fall back to the default
   return new GradleProject(<String>['debug', 'profile', 'release'], <String>[], gradleAppOutDirV1);
 }
 
+void handleKnownGradleExceptions(String exceptionString) {
+  // Handle Gradle error thrown when Gradle needs to download additional
+  // Android SDK components (e.g. Platform Tools), and the license
+  // for that component has not been accepted.
+  final String matcher =
+    r'You have not accepted the license agreements of the following SDK components:'
+    r'\s*\[(.+)\]';
+  final RegExp licenseFailure = new RegExp(matcher, multiLine: true);
+  final Match licenseMatch = licenseFailure.firstMatch(exceptionString);
+  if (licenseMatch != null) {
+    final String missingLicenses = licenseMatch.group(1);
+    if (matcher.isNotEmpty) {
```

Comment:
`matcher` is a constant string. Did you mean `missingLicenses`?

And what if `missingLicenses` is empty -- there's still a problem with the licenses (since we have a match), but we don't tell users how to fix it.